native/explorer/src/dataframe.rs

```diff
@@ -366,6 +366,16 @@ pub fn df_filter(data: ExDataFrame, mask: ExSeries) -> Result<ExDataFrame, Explo
     }
 }
 
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn df_filter_with(data: ExDataFrame, ex_expr: ExExpr) -> Result<ExDataFrame, ExplorerError> {
+    let df: DataFrame = data.resource.0.clone();
+    let exp: Expr = ex_expr.resource.0.clone();
```

Comment:
Do we need to clone? Is it mutable otherwise?

Reply:
It's because otherwise we would need to borrow, which is more complicated to handle.

The clone here is not really cloning: it's incrementing a [ResourceArc](https://docs.rs/rustler/latest/rustler/resource/struct.ResourceArc.html) for the references to this value. So this is cheap :)